### PR TITLE
Github codeowners/templates changes

### PR DESCRIPTION
**Pull request by lemunozm**
## Changes and Descriptions

- Removed `*.toml` matching to avoid having all of us in all PRs.
- (opinionated) Removed more template sections. I think most of all, do not use those sections, and in a lot of cases, it does not apply to the issue/PR purpose, and it only puts another mind barrier to open new issues. I think open issues should be something fast/easy/kind to incentivize tracking our task.

### CODE REVIEW

**Review by NunoAlexandre — DISMISSED** [review]
Agreed and approved 🤝

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,16 +1,9 @@
...
 ## Changes and Descriptions
 
 [List your changes here]
-
-# Checklist:
-
-- [ ] I have added Rust doc comments to structs, enums, traits and functions
-- [ ] I have made corresponding changes to the documentation
-- [ ] I have performed a self-review of my code
-- [ ] I have added tests that prove my fix is effective or that my feature works
```
Comment (wischli) [inline .github/pull_request_template.md:16]: I am not sure whether I can agree with this removal. Personally, I go through this checklist for every PR but I suppose the reality is, that it is rather ignored and many PRs do not include any ticks.

**Review by wischli — COMMENTED** [review]
Apart from the PR checklist removal, I fully agree with the changes. I also think the issue template is overhead it most occasions such that many chose to ignore it.

File: .github/issue_template.md [hunk 2]
```diff
@@ -1,15 +1,2 @@
...
-
-[Which codebases, pallets or designs did you base your design on]
-
-### How will this affect the code base
-
-[Does it improve readability, performance? Will it introduce new complexity?]
-
-### What are foreseen obstacles or hurdles to overcome?
-
-[Are migrations needed, structural changes around testing etc?]
```
Comment (mustermeiszer) [inline .github/issue_template.md:15]: I like that sections TBH and found it helpful to have the structure.

File: .github/pull_request_template.md (hunk 1 above)
Comment (cdamian) [reply .github/pull_request_template.md:16]: Same here, I like having this checklist for chores that might be forgotten.

File: .github/issue_template.md (hunk 2 above)
Comment (NunoAlexandre) [reply .github/issue_template.md:15]: my POV is that these are often good questions to ask oneself when creating an issue but I personally don't find value is having to try and fit this structure per se. my 2cents, whatever is more helpful to the team is good with me 👍

Comment (cdamian) [reply .github/issue_template.md:15]: IMO a section for description and probably another one for extra notes might be more than enough.

File: .github/pull_request_template.md (hunk 1 above)
Comment (lemunozm) [reply .github/pull_request_template.md:16]: Yes, In some PRs, I also found myself reading it to check mentally if things were done.

But, I found 2 points against having this there (maybe some other in place?).
- Most of the time, some points do not apply to the PR meaning.
- Most of the time, PRs are merged without clicking done in all those points.

From my point of view, I think each PR has its own checklist and should be thought of specifically for the PR purpose. So I agree with adding a checklist, but I think should be done explicitly at that moment of creation for the content of that PR.

File: .github/issue_template.md (hunk 2 above)
Comment (mustermeiszer) [reply .github/issue_template.md:15]: I am good with what the matority wants

Comment (lemunozm) [reply .github/issue_template.md:15]: I do not have a strong opinion of these sections. It's true that sometimes they do not apply to the issue or can be handled by tags (as "migration needed"), but they also help to explain the issue better

**Comment by lemunozm** [comment]
I knew it would open discussions 😆

**Review by mustermeiszer — APPROVED** [review]

**Comment by lemunozm** [comment]
I'll leave this PR open until/during next week, to get a chance for others to give arguments

File: .github/issue_template.md (hunk 2 above)
Comment (wischli) [reply .github/issue_template.md:15]: Personally, I would like to keep this but it seems like the majority choses to ignore it. I don't have a strong opinion.

Comment (NunoAlexandre) [reply .github/issue_template.md:15]: @wischli maybe we can find a middle ground that makes everyone happy?

**Comment by lemunozm** [comment]
I think, given most of you want to keep these template parts, I'm going to leave it as it is. If in some months you find it less usable, then can remove it. Is everyone ok with it?

**Comment by lemunozm** [comment]
@wischli @NunoAlexandre @cdamian @mustermeiszer would you want to add yourself to the CODEOWNERS file with the new liquidity-pools-related content we added last month?

Feel free to add a commit in this branch

**Pull request merged**